### DISCUSSION TITLE
teleop

### DISCUSSION THREAD

**Discussion by uwdal27** [Robotic-Decision-Making-Lab/blue#199] [Q&A]
Hi,

I am trying to use blue to operate a bluerov heavy with a joystick (joystick -> laptop -> blue -> rov). I have had limited success with `ros2 launch blue_bringup bluerov2_heavy.launch.yaml use_sim:=false" use_joy="true" mavros_file:="src/tag/mavros_new.yaml"`. Also, I tried the joystick instructions on [https://robotic-decision-making-lab.github.io/blue/tutorials/teleop/](https://robotic-decision-making-lab.github.io/blue/tutorials/teleop/) with `use_sim="false"` but was unable to get any reactions from the motors but I was able to get reactions on /cmd_vel and /integral_sliding_mode_controller/reference. Do you have any advice for getting this setup going? Thank you very much!

**Comment by evan-palmer** [comment]
Hey @uwdal27, thanks for checking the project out! When you say that you have had limited success with the launch file, are you referring to the mentioned motor response issue?

Regarding getting a motor response, I would anticipate that you are observing one of two things:
1. The ISMC is a feedback controller and won't send any commands to the TAM controller until it gets state feedback (velocity state). I recommend debugging this by checking the output of that controller. You can also verify that commands are being propagated through the full control architecture by checking the output of the thruster controllers.
2. The ArduSub thruster hardware interface isn't properly configuring the motor parameters to passthrough mode. You can debug this by using BlueOS or QGC to verify that, when the hardware interface is loaded, the `SERVON_FUNCTION` parameters is set to passthrough mode.

**Reply by uwdal27** [reply]
Thank you, after I fixed issue 1 things started to work! The motors seemed to have a ~1-2 second delay, is that to be expected or is there something I should work on?

**Reply by evan-palmer** [reply]
Hmm, I wouldn't expect that large of a delay in the tracking, no. When you send a command, how quickly does the ISMC respond? Typically, when I am doing test, I send my joystick input to ArduSub for manual control (using e.g., RC override) and then use the ISMC (or some other controller) for autonomous control

**Reply by uwdal27** [reply]
When I echo /mavros/rc/override, /integral_sliding_mode_controller/reference, and /cmd_vel they all seem to update simultaneous with the controller input. Is there somewhere else I should look for a bottleneck? Thank you for your help, I appreciate it!

**Reply by evan-palmer** [reply]
Do the ArduSub thruster parameters get set properly when you launch? Another debugging step worth trying is to send an override message to a single thruster (turn it on and off again quickly to avoid causing any damage) so that you can evaluate the system response there. Are you running Blue from your topside computer or from the vehicle?
